packages/core/src/storage/struct-append.test.ts

```diff
@@ -0,0 +1,135 @@
+import test from 'ava';
+import { processFile } from '../solc/compile';
+import { extractStorageLayout } from './extract';
+import { astDereferencer } from '../ast-dereferencer';
+import { StorageLayoutComparator } from './compare';
+
+async function compileAndCompare(baseCode: string, upgradedCode: string) {
```

Comment:
Can you define the test contracts in a .sol file in `packages/core/contracts/test` so that it can be built as part of the Hardhat project for tests, and change the testcases to check for specific storage upgrade errors using `getStorageUpgradeErrors` similar to https://github.com/OpenZeppelin/openzeppelin-upgrades/blob/master/packages/core/src/storage-namespaced.test.ts?